engine/classes/Elgg/ImageService.php

```diff
@@ -198,4 +199,42 @@ public function normalizeResizeParameters($width, $height, array $params = []) {
 		];
 	}
 
+	/**
+	 * Determine the image file format, this is needed for correct resizing
+	 *
+	 * @param string $filename path to the file
+	 * @param array  $params   array of resizing params (can contain 'format' to set save format)
+	 *
+	 * @see https://github.com/Elgg/Elgg/issues/10686
+	 * @return void|string
+	 */
+	protected function getFileFormat($filename, $params) {
```

Comment:
> Can we use Imagine to do this? 

I looked but couldn't find any

>We also have a similar method in one of the file related classes (see ElggFile::detectMimeType())

maybe but `$filename` doens't have to be in the Elgg file structure (eg uploaded file is in /tmp)